whelktool/scripts/2019/01/delete-spec-temporal-subjects-and-update-refs.groovy

```diff
@@ -0,0 +1,167 @@
+COMPLEX_SUBJECT_TYPE = 'ComplexSubject'
+SAO_URI = 'https://id.kb.se/term/sao'
+PREFLABEL = 'prefLabel'
+TEMPSUB_TYPE = 'TemporalSubdivision'
+SUBJECTS_TO_DELETE = ['pm14bcq70h0qnnr': 'medeltiden',
+                      '53hlst5p58lc91k': 'forntiden',
+                      '64jmtv6q2v2t5b9': 'antiken',
+                      '75knvw8r0qh4c88': 'renässansen']
+
+
+PrintWriter failedAuthIDs = getReportWriter("failed-to-delete-authIDs")
+PrintWriter scheduledForDeletion = getReportWriter("scheduled-for-deletion")
+
+
+String termToUri(term) {
+    return SAO_URI + '/' + URLEncoder.encode(term.capitalize(), "UTF-8").replaceAll("\\+", "%20")
+}
+
+String getIdOfTerm(term) {
+    String uri
+    selectBySqlWhere("""
+            collection = 'auth' AND data#>>'{@graph,1,@type}' = 'Geographic' AND data#>>'{@graph,1,prefLabel}' = '${term}'
+    """) { doc ->
+        uri = doc.graph[1][ID]
+    }
+    return uri
+}
+
+boolean updateReference(work) {
+    def extractedTerms = []
+    def entitiesToMove = []
+    def termsToChange = SUBJECTS_TO_DELETE.values() as List
+
+    if (!work.subject) return
+
+    ListIterator iterSubj = work.subject.listIterator()
+    while (iterSubj.hasNext()) {
```

Comment:
I need to use an Iterator because the script will remove the entire termComponentList entity if the termComponentList only contains one entity after removing references to 'Medeltiden', 'Antiken', 'Forntiden' and 'Renässansen'.